src/vsr/replica.zig

```diff
@@ -10365,13 +10125,40 @@ fn message_body_as_prepare_headers(message: *const Message) []const Header.Prepa
 fn message_body_as_headers_unchecked(message: *const Message) []const Header.Prepare {
     assert(message.header.size > @sizeOf(Header)); // Body must contain at least one header.
     assert(message.header.command == .do_view_change or
-        message.header.command == .start_view or
         message.header.command == .headers);
 
     return std.mem.bytesAsSlice(
         Header.Prepare,
-        message.buffer[@sizeOf(Header)..message.header.size],
+        message.body(),
+    );
+}
+
+fn start_view_message_checkpoint(message: *const Message.StartView) *const vsr.CheckpointState {
+    assert(message.header.command == .start_view);
+    const checkpoint = std.mem.bytesAsValue(
+        vsr.CheckpointState,
+        message.body()[0..@sizeOf(vsr.CheckpointState)],
     );
+    assert(checkpoint.header.valid_checksum());
+    assert(stdx.zeroed(&checkpoint.reserved));
+    return checkpoint;
+}
+
+fn start_view_message_headers(message: *const Message.StartView) []const Header.Prepare {
+    assert(message.header.command == .start_view);
+    // Body must contain at least one header.
+    assert(message.header.size > @sizeOf(Header) + @sizeOf(vsr.CheckpointState));
+
+    const headers = std.mem.bytesAsSlice(
+        Header.Prepare,
+        message.body()[@sizeOf(vsr.CheckpointState)..],
+    );
+    // To run verification.
+    _ = vsr.Headers.ViewChangeSlice.init(.start_view, headers);
+    if (constants.verify) {
+        for (headers) |header| assert(header.valid_checksum());
+    }
+    return headers;
```

Comment:
As a follow up, looks like our header parsign fucntions are no longer polymorphic, so we could perhaps could have `Message.StartView.checkpoint`, `Message.Headers.headers`, etc.